src/Cdn/Cdn/Endpoint/NewAzureRmCdnEndpoint.cs

```diff
@@ -103,6 +103,55 @@ public class NewAzureRmCdnEndpoint : AzureCdnCmdletBase
             HelpMessage = "The tags to associate with the Azure CDN endpoint.")]
         public Hashtable Tag { get; set; }
 
+        [Parameter(Mandatory = false, HelpMessage = "The default origin group.")]
+        [ValidateNotNullOrEmpty]
+        public string DefaultOriginGroup { get; set; } 
+
+        [Parameter(Mandatory = false, HelpMessage = "Azure CDN origin priority.")]
+        [ValidateNotNullOrEmpty]
+        public int? Priority { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Azure CDN origin weight.")]
+        [ValidateNotNullOrEmpty]
+        public int? Weight { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "A custom message to be included in the approval request to connect to the Private Link.")]
+        [ValidateNotNullOrEmpty]
+        public string PrivateLinkApprovalMessage { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Azure CDN origin private link location.")]
+        [ValidateNotNullOrEmpty]
+        public string PrivateLinkLocation { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Azure CDN origin private link resource id.")]
+        [ValidateNotNullOrEmpty]
+        public string PrivateLinkResourceId { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Azure CDN origin group ids.")]
+        [ValidateNotNullOrEmpty]
+        public List<string> OriginId { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The name of the origin group.")]
+        public string OriginGroupName { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The number of seconds between health probes.")]
+        [ValidateNotNullOrEmpty]
+        public int? OriginGroupProbeIntervalInSeconds { get; set; }
```

Comment:
```suggestion
        public int OriginGroupProbeIntervalInSeconds { get; set; }
```
It's discuraged to type an *optional* parameter with a nullable type. Users may wonder if there's difference between not specifying it and specify a `$null`. If the SDK requires a nullable type, you need to do the conversion.

Please apply this to all the new parameters. Thanks